wherehows-web/app/components/datasets/compliance/export-policy.ts

```diff
@@ -0,0 +1,172 @@
+   * @type {string}
+   */
+  elementId = <any>undefined;
+
+  /**
+   * Allows us to use our constants set in this enum inside our hbs template
+   * @type {ComplianceEdit}
+   */
+  ComplianceEdit = ComplianceEdit;
+
+  /**
+   * Passed in from the parent component, this determines the edit state of the compliance tab
+   * @type {boolean}
+   */
+  isEditing: boolean;
+
+  /**
+   * Passed in from the parent component, this function acts as an action to trigger edit mode on
+   * this component
+   * @type {Ember.ActionHandler}
+   */
+  toggleEditing: (edit: boolean, target: ComplianceEdit) => void;
+
+  /**
+   * Passed in from the parent component, this function triggers the saving logic and subsequent POST
+   * request to persist the export policy edits
+   * @type {Ember.ActionHandler}
+   */
+  onSaveExportPolicy: (policy: IDatasetExportPolicy) => void;
+
+  /**
+   * User triggered flag in which they desire to see all rows of the export policy by clicking "show more"
+   * @type {boolean}
+   * @memberof ComplianceExportPolicy
+   */
+  shouldShowMorePolicyData = false;
+
+  /**
+   * Whether we should show all of the export policy rows. Based on whether we are editing or user desires
+   * to see all the rows
+   * @type {boolean}
+   * @memberof ComplianceExportPolicy
+   */
+  shouldShowAllExportPolicyData = or('isEditing', 'shouldShowMorePolicyData');
+
+  /**
+   * The export policy data extracted directly from the api response, passed in from the dataset-compliance
+   * container and stored here as its original response
+   * @type {IDatasetExportPolicy}
+   * @memberof ComplianceExportPolicy
+   */
+  exportPolicyData: IDatasetExportPolicy;
+
+  /**
+   * Calculates the table for the export policy questions with seeded information based on the api response
+   * for the dataset's export policy
+   * @type {ComputedProperty<Array<IExportPolicyTable>>}
+   * @memberof ComplianceExportPolicy
+   */
+  datasetExportPolicy = computed('exportPolicyData', 'isEditing', function(): Array<IExportPolicyTable> {
+    const exportPolicyData: IDatasetExportPolicy = getWithDefault(this, 'exportPolicyData', {});
+
+    return policyKeys.map(key => {
+      const dataType = ExportPolicyKeys[key];
+
+      return {
+        dataType,
+        value: exportPolicyData[dataType] || false,
+        label: ExportPolicyLabels[key]
+      };
+    });
+  });
+
+  constructor() {
+    super(...arguments);
+    // Setting our default values for properties expected from parent
+    typeof this.isEditing === 'boolean' || (this.isEditing = false);
+    this.toggleEditing || (this.toggleEditing = noop);
+    this.onSaveExportPolicy || (this.onSaveExportPolicy = noop);
+  }
+
+  /**
+   * Action to mark all fields in our working copy as false for the export policy
+   */
+  @action
+  onMarkAllExportFieldsAsFalse(this: ComplianceExportPolicy): void {
+    const exportPolicy = get(this, 'datasetExportPolicy');
+
+    exportPolicy.forEach(policy => {
```

Comment:
Can you make this a non mutative operation? so assign exportPolicy with a reduction instead. Also, isn't `datasetExportPolicy` a computed property? seems weird to be changing the values on a computed property directly...

Reply:
Also changing the values on it was just a cute way I found to reset back if the user cancels the change. It prevents us from having to save an "original" copy before they make changes in the UI. Since the exportPolicyData is the source of truth and does not change, if the user cancels, they trigger another computation here and it reverts back. If they hit save, it triggers a change in the exportPolicyData and triggers a recomputation here to persist the changes.

Reply:
With reverting back, is it possible that the previous value was `true`? What if the user was making the change on a previously persisted policy that had a `true` value and decided to cancel, won't this incorrectly set it to `false` or  am I missing some context?

Reply:
As per the former, mutation, yes but what I mean is for example using Object#assign in place of directly setting the value of the object key? if `datasetExportPolicy` is a CP it will require a setter as opposed to a direct mutation